### PR TITLE
fix(ui-library): align counter to right for text area with or without error

### LINKED ISSUE #1064: Text Area - Counter gets left-aligned if hasError=true
### What is affected?

Component

### Description

When I open the Text Area component in Storybook and set "hasError" to true (with default values), the counter gets left aligned but I expect the counter to still be right aligned.

Please also add a test to make sure the counter is always placed as expected, we had similar issues with counter placement before.

### Steps to reproduce

1. [Open Text Area component in storybook](https://b01ler.onrender.com/?path=/story/components-text-area--text-area)
2. Set "hasError" to true (leave everything else as defuault (hasCounter=true, errorMessage=empty, errorMessageIcon=undefined)
3. See that the Counter switches from being right aligned to being left aligned

### Affected version(s)

current develop version

### Operating system / Browser / Environment

Chrome Browser

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ashk1996**
closes #1064

### CODE REVIEW

File: packages/ui-library/src/components/textarea/index.css.ts [hunk 1]
```diff
@@ -34,11 +34,15 @@ export const staticStyles = typeSafeNestedCss/*css*/ `
   .blr-textarea-info-container {
     display: flex;
     justify-content: right;
-
-    &.hint, &.error {
-      justify-content: space-between;
-    }
   }
+  
+  .blr-textarea-info-container.error {
```
Comment (angsherpa456) [inline packages/ui-library/src/components/textarea/index.css.ts:39]: For consistency, please follow the similar pattern like other:
.blr-textarea-info-container {
&.error {...}
&.hit {...}
}

File: packages/ui-library/src/components/textarea/index.css.ts [hunk 2]
```diff
@@ -34,11 +34,13 @@ export const staticStyles = typeSafeNestedCss/*css*/ `
...
-
-    &.hint, &.error {
+    &.error{
+      justify-content: right;
+    }
+    &.hint{
       justify-content: space-between;
     }
-  }
+  }
```
Comment (angsherpa456) [inline packages/ui-library/src/components/textarea/index.css.ts:43]: Seems like your branch is out of sync with develop branch, please fix that and will approve it. Also please remove this extra space at line 43.

**Review by angsherpa456 — APPROVED** [review]
LGTM

**Review by thrbnhrtmnn — CHANGES_REQUESTED** [review]
I just had a quick look and found one issue. If you add an error message and error message icon, they are right aligned, but they should be left aligned. 
Screenshot 2024-05-23 at 09 20 46

Here is how it should look according to design: 
Screenshot 2024-05-23 at 09 24 36

**Comment by ashk1996** [comment]
> I just had a quick look and found one issue. If you add an error message and error message icon, they are right aligned, but they should be left aligned. Screenshot 2024-05-23 at 09 20 46
> 
> Here is how it should look according to design: Screenshot 2024-05-23 at 09 24 36

Hi @thrbnhrtmnn 
Just doubled checked, I see that it is left aligned. 
Screenshot 2024-05-23 at 10 08 13

**Comment by thrbnhrtmnn** [comment]
> > I just had a quick look and found one issue. If you add an error message and error message icon, they are right aligned, but they should be left aligned. Screenshot 2024-05-23 at 09 20 46
> > Here is how it should look according to design: Screenshot 2024-05-23 at 09 24 36
> 
> Hi @thrbnhrtmnn Just doubled checked, I see that it is left aligned. Screenshot 2024-05-23 at 10 08 13

Hey @ashk1996 , in your example hasHint = true. In this case it looks good. But if hasHint = false and only hasError = true and errorMessage is not empty, then the error message is not left aligned but right aligned, as seen in my screenshot above. 

While looking at it again, if errorMessageIcon is not undefined and errorMessage is empty, the icon should also be displayed right aligned, but currently it is not displayed at all in this case. It will only appear once errorMessage is not empty anymore.

If you want we can have a quick huddle about this :-)

**Comment by ashk1996** [comment]
> > > I just had a quick look and found one issue. If you add an error message and error message icon, they are right aligned, but they should be left aligned. Screenshot 2024-05-23 at 09 20 46
> > > Here is how it should look according to design: Screenshot 2024-05-23 at 09 24 36
> > 
> > 
> > Hi @thrbnhrtmnn Just doubled checked, I see that it is left aligned. Screenshot 2024-05-23 at 10 08 13
> 
> Hey @ashk1996 , in your example hasHint = true. In this case it looks good. But if hasHint = false and only hasError = true and errorMessage is not empty, then the error message is not left aligned but right aligned, as seen in my screenshot above.
> 
> While looking at it again, if errorMessageIcon is not undefined and errorMessage is empty, the icon should also be displayed right aligned, but currently it is not displayed at all in this case. It will only appear once errorMessage is not empty anymore.
> 
> If you want we can have a quick huddle about this :-)

Thanks for pointing out. I have fixed this in the latest commit.

**Comment by thrbnhrtmnn** [comment]
Hey @ashk1996 , as discussed today the icon not being displayed if the corresponding message is empty is not an issue, because this would just be a user mistake. There should not be an icon when there is no message. Neither for errors nor for hints.

After looking again the only point I found is that in case the hasHint = true, but message and icon are empty / undefined, the counter will be left aligned. As just discussed we should fix this, as long it is not a huge effort. Again this would be an user mistake, but if we can prevent it easily we should always do so.

**Review by thrbnhrtmnn — APPROVED** [review]

**Pull request merged**